TypeCobol/Compiler/Diagnostics/Cobol2002Checker.cs

```diff
@@ -27,29 +28,33 @@ public static void CheckRedefines(DataRedefinesEntry redefines, CodeElementsPars
 
         public static void CheckTypedef(DataTypeDescriptionEntry typedef, CodeElementsParser.DataDescriptionEntryContext context)
         {
+            var firstToken = ParseTreeUtils.GetFirstToken(context.cobol2002TypedefClause());
+            var copyDirective = (firstToken as ImportedToken)?.CopyDirective;
+            string copyErrorMessage = copyDirective == null ? "" : $" in '{copyDirective.TextName}' copy";
+
             if (typedef.LevelNumber?.Value != 1)
             {
-                string message = "TYPEDEF clause can only be specified for level 01 entries";
+                string message = "TYPEDEF clause can only be specified for level 01 entries" + copyErrorMessage;
                 DiagnosticUtils.AddError(typedef, message, context.cobol2002TypedefClause());
             }
 
             if (typedef.IsExternal)
             {
-                string message = "EXTERNAL clause cannot be specified with TYPEDEF clause";
+                string message = "EXTERNAL clause cannot be specified with TYPEDEF clause" + copyErrorMessage;
                 foreach (var external in context.externalClause())
                     DiagnosticUtils.AddError(typedef, message, external);
             }
 
             if (typedef.HasExplicitVisibility && typedef.IsGlobal)
             {
-                string message = $"GLOBAL clause cannot be specified with {typedef.Visibility.ToString().ToUpper()} access modifier";
+                string message = $"GLOBAL clause cannot be specified with {typedef.Visibility.ToString().ToUpper()} access modifier" + copyErrorMessage;
```

Comment:
From An API perspective, the COPY name shold be a separate parameter of `DiagnosticUtils.AddError`.
It will also allow to change the positon of the copy name inside the error message easily.

Or even better, can the COPY name could be retrieved automatically within the AddError method.

Reply:
done in AddError method.
https://github.com/TypeCobolTeam/TypeCobol/pull/1602/commits/96bff72e166deda2c10a40bb4a19f899b6c4dfdd